### ISSUE TITLE
MaaS Horizon check fails when using custom branding

### ISSUE THREAD

**Issue by JCallicoat** [rcbops/rpc-openstack#1135]
##### What version (commit) do you encounter this issue on?

r12.1.1

##### What type of issue is this (bug, feature, documentation, regression)?

Bug

##### What is the expected behavior?

The horizon check does not fail when using custom branding

##### What is actually happening?

The horizon check returns "could not load login page" when using custom branding.

##### What were you attempting to do when you encountered this error (give as much detail as possible)? 

Set `horizon_site_name` in user_variables.yml and run os-horizon-install.yml which will set `SITE_BRANDING` in /etc/horizon/local_settings.py, overriding the default `SITE_BRANDING = 'OpenStack Dashboard'` in /openstack_dashboard/settings.py.

The horizon check will do a regex search for 'openstack dashboard' in the response body and fail.

https://github.com/rcbops/rpc-openstack/blob/r12.1.1/maas/plugins/horizon_check.py#L61-L63

##### How can we reproduce this issue?

Follow steps above

##### How can we verify your issue is addressed?

Run horizon check and verify metrics are returned

##### What is your evaluation on the importance of this issue, and why?

Low impact, most customers do not use custom branding at this time

###### Remove branches (lines) you do not wish to target.
- [x] Master (#1255)
- [x] liberty-12.2 (#1264)
- [x] kilo (#1265)

**Comment by b3rnard0** [comment]
- [ ] Reproduce the issue. (N/A)
- [ ] Create a PR for fix and/or docs.
- [ ] Incorporate feedback from pull request reviews.
- [ ]  What is the gating for this on DEV and/or QE? Add related gating if appropriate. Also make sure current gating passes with this fix.
- [ ] Reach out to reporter to determine if the fix is resolving the issue.
- [ ] What are the upgrade impacts (if any).
- [ ] What are the backports are needed (if any).

**Comment by ameserole** [comment]
UPDATE:
Running into some issues getting an r12.1.1 environment set up to try and reproduce the bug. Tomorrow we will see if we can get an r12.1.1 lab set up. We will try and reproduce it on master as well.

**Comment by ameserole** [comment]
UPDATE:
We have reproduced the issues and are now working on a solution.

**Comment by ameserole** [comment]
UPDATE:
Submitted a work in progress pull request. The fix has not been fully tested yet.

**Comment by alextricity25** [comment]
Update: The pull request is ready for review.

@JCallicoat Is the process of updating the horizon_site_name documented anywhere? Or would you like us to submit a doc patch outlining the process of changing horizon_site_name? If the horizon_site_name wants to be changed on an existing environment, the new process will involve re-running setup-maas after re-running the os-install-horizon playbooks.

**Comment by JCallicoat** [comment]
@alextricity25 I'm not sure tbh. It used to be documented when it was part of an external repository.

https://github.com/evrardjp/ansible-openstack-horizon-branding/blob/617bf88395909b683d571f87ead798f24a439b9f/README.md

When that was merged over it looks like the documentation was removed in favor of describing it in the commit message?

https://github.com/openstack/openstack-ansible/commit/5c9594d4db8246759c917bef918e53fc26a87e80

But maybe it was split off to a docs repository? I don't really know.

**Comment by ameserole** [comment]
UPDATE:
Looks like #1255 is going to get approval to be merged soon once it passes jenkins.
Working on backports for kilo and liberty now.

**Comment by ameserole** [comment]
UPDATE:
Added pull requests for backports. #1264 for liberty and #1265 for kilo

**Comment by git-harry** [comment]
@ameserole please don't create custom labels, you seem to have added the label 'in progress' which is not defined in the process that was agreed Friday before last. I will remove this label, please let me know what you were trying to achieve with it and we can explore how that fits in with current expectations.

**Comment by ameserole** [comment]
@git-harry to be honest I don't know how those got there. I noticed that it showed up when I made on the backports when I made them but never did anything about it. I will keep this in mind for the future.

**Comment by git-harry** [comment]
@ameserole that's fine, I just want to make sure that when we add labels they have a clear purpose and we document it.

**Comment by ameserole** [comment]
UPDATE:
#1255 merged into master and now just waiting on the backports #1265 and #1264 to be reviewed and merged.

**Comment by ameserole** [comment]
UPDATE:
#1265 and #1264 have now both merged.
Im going to close this issue unless @JCallicoat has anymore problems with this.

**Comment by JCallicoat** [comment]
LGTM :+1:
